src/views/imaged-object-editor/ImageMenu.vue

```diff
@@ -0,0 +1,359 @@
+                </div>
+                <div class="col">          
+                  <b-form-input type="range" min="2" max="40" step="1"  v-model="brushSize"></b-form-input>
+                </div>
+              </div>
+            </section>
+            <section v-if="artefact.mask">
+              <b-form-checkbox v-model="mask">Mask</b-form-checkbox>
+            </section>
+          </b-card-body>
+        </b-collapse>
+      </b-card>
+    </section>
+
+    <section>
+      <b-card no-body class="mb-1">
+        <b-card-header header-tag="header" class="p-1" role="tab">
+          <b-button block href="#" v-b-toggle.accordion-actions variant="info">Actions</b-button>
+        </b-card-header>
+        <b-collapse id="accordion-actions" accordion="my-accordion" role="tabpanel">
+          <b-card-body>
+            <section class="center-btn">
+              <b-button @click="onRotateClick(-90)"><font-awesome-icon icon="undo"></font-awesome-icon></b-button>
+              <b-button @click="onRotateClick(90)"><font-awesome-icon icon="redo"></font-awesome-icon></b-button>
+            </section>
+            <section class="center-btn" v-if="editable">
+                <b-button v-for="mode in [{name: 'Draw', val:'DRAW'}, {name: 'Erase', val: 'ERASE'}]" 
+                :key="mode.val" @click="onDrawChanged(mode.val)" 
+                :pressed="modeChosen(mode.val)">{{ mode.name }}</b-button>
+            </section>
+            <section class="center-btn" v-if="editable" v-shortcuts="[
+              { shortcut: [ 'arrowright' ], callback: redoModal },
+              { shortcut: [ 'arrowleft' ], callback: undoModal },
+            ]">
+            
+              <b-button @click="undo()">Undo</b-button>
+              <b-button @click="redo()">Redo</b-button>
+            </section>
+            <section class="center-btn" v-if="editable">
+              <b-button  v-if="!saving" @click="save()">Save</b-button>
+              <b-button v-if="saving" disabled class="disable">
+                Saving...<font-awesome-icon icon="spinner" size="1.5x" spin></font-awesome-icon>
+              </b-button>
+            </section>
+          </b-card-body>
+        </b-collapse>
+      </b-card>
+    </section>
+</div>
+</template>
+
+<script lang="ts">
+import Vue, { PropOptions } from 'vue';
+import { ImagedObject } from '@/models/imaged-object';
+import { Artefact } from '@/models/artefact';
+import { EditorParams, DrawingMode, EditorParamsChangedArgs, SingleImageSetting, OptimizedArtefact } from './types';
+import SingleImageSettingComponent from './SingleImageSetting.vue';
+import ImagedObjectService from '../../services/imaged-object';
+/**
+ * This component has a lot of emit functions.  Perhaps it will be better
+ * to create a modular container that holds this menu and the possible
+ * image view modules that can accompany it.  If we want the menu to be
+ * a component, however, I see no way around using these emit functions.
```

Comment:
Yeah, we may be working a bit against the Vue design style here.  Your suggestion is good.  An event bus is another possible design scenario.

Reply:
I suggest we do this later and not now.

Reply:
Agreed.